StarRecordImpl.newCoreRecord is incompatible with StarRecordImpl.hasExtension

StarRecordImpl.newCoreRecord clears the contents of lists of extensions, but leaves the keys in the map. This causes StarRecordImpl.hasExtension to believe that extensions of previously existing row types may still exist. 

StarRecordImpl.newCoreRecord should either remove the keys also, or hasExtension should check the list is non-empty before returning true.

I am having issues compiling with 3 of the gbif components that are dependencies of dwca-io, but not sure where to file an issue about it.

I tried to debug it by updating the plugin versions in a local version of motherpom, but not sure what to do with the wagon versions as since Wagon was transferred into Apache it isn't released as maven-wagon-plugin anymore. I don't remember having any good experiences with Wagon in the past, but you tend to remember bad experiences more than good experiences.

```
Downloading: http://repository.gbif.org/content/groups/gbif/org/gbif/dwc-api/1.16/dwc-api-1.16.pom
Downloading: http://repository.gbif.org/content/groups/gbif/org/gbif/gbif-common/0.29/gbif-common-0.29.pom
Downloading: http://repository.gbif.org/content/groups/gbif/org/gbif/registry/registry-metadata/2.59/registry-metadata-2.59.pom
```

The error only refers to dwc-api, but I am getting the same issues with gbif-common and registry-metadata, but not with motherpom for some reason.

```
Caused by: org.eclipse.aether.transfer.ArtifactTransferException: Could not transfer artifact org.gbif:dwc-api:pom:1.16 from/to gbif-all (http://repository.gbif.org/content/groups/gbif): GET request of: org/gbif/dwc-api/1.16/dwc-api-1.16.pom from gbif-all failed
	at org.eclipse.aether.connector.basic.ArtifactTransportListener.transferFailed(ArtifactTransportListener.java:43)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run(BasicRepositoryConnector.java:355)
	at org.eclipse.aether.util.concurrency.RunnableErrorForwarder$1.run(RunnableErrorForwarder.java:67)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$DirectExecutor.execute(BasicRepositoryConnector.java:581)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector.get(BasicRepositoryConnector.java:249)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.performDownloads(DefaultArtifactResolver.java:520)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:421)
	... 36 more
Caused by: org.apache.maven.wagon.TransferFailedException: GET request of: org/gbif/dwc-api/1.16/dwc-api-1.16.pom from gbif-all failed
	at org.apache.maven.wagon.AbstractWagon.getTransfer(AbstractWagon.java:363)
	at org.apache.maven.wagon.AbstractWagon.getTransfer(AbstractWagon.java:316)
	at org.apache.maven.wagon.AbstractWagon.getTransfer(AbstractWagon.java:285)
	at org.apache.maven.wagon.StreamWagon.getIfNewer(StreamWagon.java:97)
	at org.apache.maven.wagon.StreamWagon.get(StreamWagon.java:61)
	at org.eclipse.aether.transport.wagon.WagonTransporter$GetTaskRunner.run(WagonTransporter.java:560)
	at org.eclipse.aether.transport.wagon.WagonTransporter.execute(WagonTransporter.java:427)
	at org.eclipse.aether.transport.wagon.WagonTransporter.get(WagonTransporter.java:404)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$GetTaskRunner.runTask(BasicRepositoryConnector.java:447)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run(BasicRepositoryConnector.java:350)
	... 41 more
Caused by: java.net.SocketException: Connection reset
	at java.net.SocketInputStream.read(SocketInputStream.java:209)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at org.apache.maven.wagon.providers.http.httpclient.impl.io.SessionInputBufferImpl.streamRead(SessionInputBufferImpl.java:136)
	at org.apache.maven.wagon.providers.http.httpclient.impl.io.SessionInputBufferImpl.fillBuffer(SessionInputBufferImpl.java:152)
	at org.apache.maven.wagon.providers.http.httpclient.impl.io.SessionInputBufferImpl.read(SessionInputBufferImpl.java:203)
	at org.apache.maven.wagon.providers.http.httpclient.impl.io.ContentLengthInputStream.read(ContentLengthInputStream.java:178)
	at org.apache.maven.wagon.providers.http.httpclient.conn.EofSensorInputStream.read(EofSensorInputStream.java:137)
	at org.apache.maven.wagon.AbstractWagon.transfer(AbstractWagon.java:531)
	at org.apache.maven.wagon.AbstractWagon.getTransfer(AbstractWagon.java:353)
	... 50 more
```

I get the same Connection Reset errors with wget (ie, outside of maven land) but it retries and succeeds:

```
$ wget http://repository.gbif.org/content/groups/gbif/org/gbif/dwc-api/1.16/dwc-api-1.16.pom
--2016-11-22 10:50:12--  http://repository.gbif.org/content/groups/gbif/org/gbif/dwc-api/1.16/dwc-api-1.16.pom
Resolving repository.gbif.org (repository.gbif.org)... 130.226.238.146
Connecting to repository.gbif.org (repository.gbif.org)|130.226.238.146|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 2996 (2.9K) [application/xml]
Saving to: 'dwc-api-1.16.pom.1’

dwc-api-1.16.pom.1                                                               83%[====================================================================================================================================================================>                                 ]   2.44K  --.-KB/s    in 0s      

2016-11-22 10:50:13 (227 MB/s) - Read error at byte 2499/2996 (Connection reset by peer). Retrying.

--2016-11-22 10:50:14--  (try: 2)  http://repository.gbif.org/content/groups/gbif/org/gbif/dwc-api/1.16/dwc-api-1.16.pom
Connecting to repository.gbif.org (repository.gbif.org)|130.226.238.146|:80... connected.
HTTP request sent, awaiting response... 206 Partial Content
Length: 2996 (2.9K), 497 remaining [application/xml]
Saving to: 'dwc-api-1.16.pom.1’

dwc-api-1.16.pom.1                                                              100%[+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++================================>]   2.93K  --.-KB/s    in 0s      

2016-11-22 10:50:15 (105 MB/s) - 'dwc-api-1.16.pom.1’ saved [2996/2996]
```

Curl fails with connection reset by peer at the same number of bytes.

--- a/.gitignore
+++ b/.gitignore
@@ -8,3 +8,7 @@ target/
 *.jar
 *.war
 *.ear
+
+.classpath
+.project
+.settings/

--- a/pom.xml
+++ b/pom.xml
@@ -4,7 +4,7 @@
   <parent>
     <groupId>org.gbif</groupId>
     <artifactId>motherpom</artifactId>
-    <version>37</version>
+    <version>38-SNAPSHOT</version>
   </parent>
 
   <artifactId>dwca-io</artifactId>

--- a/src/main/java/org/gbif/dwca/record/StarRecordImpl.java
+++ b/src/main/java/org/gbif/dwca/record/StarRecordImpl.java
@@ -37,7 +37,7 @@ public class StarRecordImpl implements StarRecord {
   }
 
   public boolean hasExtension(Term rowType) {
-    return extensions.containsKey(rowType);
+    return extensions.containsKey(rowType) && !extensions.get(rowType).isEmpty();
   }
 
   /**

